opentelemetry-exporter-geneva/geneva-uploader/src/bench.rs

```diff
@@ -43,4 +46,218 @@ mod benchmarks {
+                ),
+            }),
+            ..Default::default()
+        }
+    }
+
+    fn create_log_with_attributes(
+        timestamp: u64,
+        num_attributes: usize,
+        rng: &mut StdRng,
+    ) -> LogRecord {
+        let mut log = LogRecord {
+            observed_time_unix_nano: timestamp,
+            severity_number: 9,
+            severity_text: "INFO".to_string(),
+            event_name: format!("Event_{}Attrs", num_attributes),
+            body: Some(AnyValue {
+                value: Some(
+                    opentelemetry_proto::tonic::common::v1::any_value::Value::StringValue(
+                        "Log with attributes".to_string(),
+                    ),
+                ),
+            }),
+            ..Default::default()
+        };
+
+        // Add attributes with different types
+        for i in 0..num_attributes {
+            let key = format!("attr_{}", i);
+            let value = match i % 4 {
+                0 => AnyValue {
+                    value: Some(
+                        opentelemetry_proto::tonic::common::v1::any_value::Value::StringValue(
+                            format!("string_value_{}", rng.gen::<u32>()),
+                        ),
+                    ),
+                },
+                1 => AnyValue {
+                    value: Some(
+                        opentelemetry_proto::tonic::common::v1::any_value::Value::IntValue(
+                            rng.gen_range(0..1000),
+                        ),
+                    ),
+                },
+                2 => AnyValue {
+                    value: Some(
+                        opentelemetry_proto::tonic::common::v1::any_value::Value::DoubleValue(
+                            rng.gen_range(0.0..100.0),
+                        ),
+                    ),
+                },
+                _ => AnyValue {
+                    value: Some(
+                        opentelemetry_proto::tonic::common::v1::any_value::Value::BoolValue(
+                            rng.gen_bool(0.5),
+                        ),
+                    ),
+                },
+            };
+
+            log.attributes.push(KeyValue {
+                key,
+                value: Some(value),
+            });
+        }
+
+        log
+    }
+
+    /*
+    - Criterion benchmarks (encode_log_batch, OtlpEncoder):
+        Results: (WSL2 Linux, 62 GiB RAM, 8-core/16-thread Intel CPU, kernel 6.6.36.3)
+        - Attribute count scaling (10 log records each):
+            - 0 attrs:   ~10.3 µs/op
+            - 4 attrs:   ~15.7 µs/op
+            - 8 attrs:   ~24.3 µs/op
+            - 16 attrs:  ~44.1 µs/op
+
+        - Batch size scaling (each log with 4 attributes):
+            - 1 log:     ~1.63 µs/op
+            - 10 logs:   ~16.4 µs/op
+            - 100 logs:  ~161 µs/op
+            - 1000 logs: ~1.59 ms/op
+
+        - Schema cache effectiveness (100 log records, each with 4 attributes, all same schema):
```

Comment:
This is already covered by the previous benchmark for Batch size scaling. We could probably remove this and update the comment about the previous benchmark to indicate that it is being tested with the same schema.

Reply:
This was specifically testing the scenario where cache is already warmed up. But I realize this was not really required. Have removed it now,